src/components/pagination/index.tsx

```diff
@@ -0,0 +1,259 @@
+	const pagination = usePagination()
+	const start = pagination.page * pagination.pageSize - pagination.pageSize + 1
+	const end = Math.min(
+		pagination.page * pagination.pageSize,
+		pagination.totalItems
+	)
+	return (
+		<div className={s.info}>
+			{start} - {end}
+			{showTotalItems ? ` of ${pagination.totalItems}` : null}
+		</div>
+	)
+}
+
+const Nav = ({ type = 'compact' }: NavProps) => {
+	const pagination = usePagination()
+
+	const totalPages = pagination.totalPages
+	const currentPage = pagination.page
+
+	const rawitems = useMemo(
+		() => Array.from({ length: totalPages }, (_, i) => i + 1),
+		[totalPages]
+	)
+
+	const items: (number | 'ellipsis')[] = ((t: NavProps['type']) => {
+		switch (t) {
+			case 'numbered':
+				return rawitems
+			case 'truncated':
+				return generateTruncatedList(rawitems, currentPage)
+			case 'compact':
+			default:
+				return []
+		}
+	})(type)
+
+	return (
+		<nav className={s.nav} aria-label="Pagination navigation">
+			<ButtonArrow type={type} direction="prev" />
+			{type !== 'compact' && (
+				<ul className={s['page-list']}>
+					{items.map((e, i) => {
+						if (e === 'ellipsis') {
+							// This key will be different on every re-render which is OK
+							const key = `${e}-${i}`
+							return (
+								<li className={classNames(s['page-item'])} key={key}>
+									<div className={s.ellipsis}>...</div>
+								</li>
+							)
+						}
+
+						const isSelected = e === currentPage
+						return (
+							<li
+								className={classNames(s['page-item'], {
+									[s['is-selected']]: isSelected,
+								})}
+								// This key will be stable between re-renders
+								// which prevents react from removing focus from
+								// the previous `document.activeElement`
+								key={e}
+							>
+								<ButtonNumber page={e} />
+							</li>
+						)
+					})}
+				</ul>
+			)}
+			<ButtonArrow type={type} direction="next" />
+		</nav>
+	)
+}
+
+const ButtonArrow = ({ type, direction }: ButtonArrowProps) => {
+	const pagination = usePagination()
+	const { handleClick, icon, isDisabled, label, className } = {
+		next: {
+			handleClick: () => {
+				// clamp upper bound
+				const nextIdx = Math.min(pagination.page + 1, pagination.totalPages)
+				pagination.setPage(nextIdx)
+			},
+			icon: <IconChevronRight16 />,
+			label: 'Next',
+			isDisabled: pagination.page === pagination.totalPages,
+			className: s.next,
+		},
+		prev: {
+			handleClick: () => {
+				// clamp lower bound (0)
+				const prevIdx = Math.max(pagination.page - 1, 0)
+				pagination.setPage(prevIdx)
+			},
+			icon: <IconChevronLeft16 />,
+			label: 'Previous',
+			isDisabled: pagination.page === 1,
+			className: s.prev,
+		},
+	}[direction]
+
+	return (
+		<button
+			className={classNames(s.arrow, s.control, className)}
+			aria-label={label}
+			onClick={handleClick}
+			disabled={isDisabled}
+		>
+			{icon}
+			{type === 'compact' && (
+				<span className={classNames(s.label)}>{label}</span>
+			)}
```

Comment:
🙋🏻 Just noticing this -- should the `label` be shown if the type is `"compact"`? I'd imagine that the label would _not_ show in this case, to save space.

Reply:
Based on the [HDS implementation](https://hds-components-git-hds-943-pagination-component-hashicorp.vercel.app/components/pagination#how-to-use), yea the "Previous/Next" label will be shown _only_ on `compact`

Reply:
Ohhh, the `type` passed in to `ButtonArrow` in the pagination type! I thought it was just scoped to the button itself. Thanks! 😊